src/readonly-table/index.ts

```diff
@@ -1,51 +1,61 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable kuc-v1/no-using-event-handler-name */
 /* eslint-disable kuc-v1/validator-in-should-update */
-import { html, PropertyValues } from "lit";
-import { property } from "lit/decorators.js";
+import { html, svg, PropertyValues } from "lit";
+import { property, query, state } from "lit/decorators.js";
 import { KucBase } from "../base/kuc-base";
 import { visiblePropConverter } from "../base/converter";
 import { validateProps } from "../base/validator";
+import { BaseLabel } from "../base/label";
+export { BaseLabel };
 
-type Column = { header?: { text?: string }; visible?: boolean };
+type Column = { headerName?: string; visible?: boolean; width?: string };
 type ReadOnlyTableProps = {
   className?: string;
   id?: string;
   label?: string;
-  visible?: boolean;
   columns?: Column[];
   data?: string[][];
+  pagination?: boolean;
+  rowsPerPage?: number;
+  visible?: boolean;
 };
 
 export class ReadOnlyTable extends KucBase {
   @property({ type: String, reflect: true, attribute: "class" }) className = "";
   @property({ type: String, reflect: true, attribute: "id" }) id = "";
   @property({ type: String }) label = "";
+  @property({ type: Array }) columns: Column[] = [];
+  @property({ type: Array }) data: string[][] = [];
+  @property({ type: Boolean }) pagination = true;
+  @property({ type: Number }) rowsPerPage = 5;
   @property({
     type: Boolean,
     attribute: "hidden",
     reflect: true,
     converter: visiblePropConverter,
   })
   visible = true;
-  @property({ type: Array }) columns: Column[] = [];
-  @property({ type: Array }) data: string[][] = [];
+  @state()
+  private _pagePosition: number = 1;
+
+  @query(".kuc-readonly-table__pager__pagenation-prev")
+  private _prevButtonEl!: HTMLButtonElement;
+  @query(".kuc-readonly-table__pager__pagenation-next")
+  private _nextButtonEl!: HTMLButtonElement;
 
   constructor(props?: ReadOnlyTableProps) {
     super();
-    if (!props) {
+    if (props && props.columns && props.data) {
+      this._validateColumns(props.columns);
+      this._validateData(props.data);
+    } else {
       return;
     }
-    if (!Array.isArray(props.columns) && props.columns !== undefined) {
-      throw new Error("'columns' property is invalid");
-    }
-    if (!Array.isArray(props.data) && props.data !== undefined) {
-      throw new Error("'data' property is invalid");
+
+    if (props.rowsPerPage || props.rowsPerPage === 0) {
```

Comment:
in my opinion , props.rowsPerPage === 0 doesn't necessary
Because  in this._validateRowsPerPage you validated when rowsPerPage equal 0